rfc/0012-policy-report.md

```diff
@@ -95,12 +96,429 @@ results from Kubernetes policy engines (like Kubewarden or Kyverno) but can
+to the `PolicyReport` are equally relevant for the `ClusterPolicyReport`.
+
+## Report
+
+When auditing a resource, certain Kubewarden policies may be relevant to the
+events occurring within it. Additionally, the same Kubernetes resource might
+undergo evaluation by multiple policies, resulting in several policy
+evaluations being conducted
+
+The `PolicyReport` provides these information:
+
+- The number of policy evaluations conducted during the audit run.
+- The count of policy evaluations resulting in resource acceptance, rejection,
+  or failure due to errors. Policy failures occur when errors arise during
+  evaluation, such as runtime panics.
+- Comprehensive details regarding all evaluations performed, including the
+  policy utilized, the evaluated resource, the final outcome (acceptance,
+  rejection, or failure), and any accompanying messages returned by the policy.
+
+
+The `PolicyReport` is kept updated during the lifetime of the audited resource,
+meaning:
+
+- New Policies might become "interested" about the resource. New evaluations
+  results will be added to the `PolicyReport`
+- Existing Kubernetes objects and policies can change over time: some
+  evaluations could change their outcome (pass -> fail and vice versa). The
+  evaluation results must reflect that
+- Policies might be deleted/might be no longer "interested" in the resource:
+  the report must be updated to not container any reference the these policies
+  and their evaluation results
+- Kubernetes resources might be deleted: the report should be deleted together
+  with if
+
+**Note:** `PolicyReport` resources contain security/compliance information. Regular
+users of the cluster should only have READ access to them. They should not be
+able to create, delete or change their contents. This can be done with Kubernetes
+RBAC rules.
+
+### `PolicyReport` detailed overview
+
+For each Namespace, Kubewarden will find all the policies that are interested
+about actions happening inside of it. This is going to be a mix of
+some `ClusterAdmissionPolicy` policies and all the `AdmissionPolicy` policies
+that are defined inside of the Namespace.
+
+Next, this list of policies is going to be filtered: only policies interested
+in `CREATE` events are going to be considered.
+The audit report is not going to simulate `DELETE` and `UPDATE` events.
+
+For each relevant policy, Kubewarden will find all the Kubernetes resources that
+are 1) defined inside of the Namespace and 2) are relevant to the policy. Kubewarden
+will then simulate the `CREATE` event of this resource and keep track of the policy
+evaluation result.
+
+All the results are going to be collected into the `PolicyReport` object
+defined inside of the Namespace. There's going to be only one `PolicyReport`
+per audited resource.
+
+The `PolicyReport` resource has the following fields:
+
+- `apiVersion` and `kind`: these are defined by the CRDs
+- `metadata`: this is the usual `meta/v1.ObjectMeta` resource. In the
```

Comment:
we added an OwnerReference in the metadata pointing to the resource to allow the report to be garbage collected when the resource is deleted